src/Atmos/Parameterizations/CloudPhysics/Microphysics.jl

```diff
@@ -0,0 +1,187 @@
+"""
+    one-moment bulk Microphysics scheme
+
+Microphysics parameterization based on the ideas of Kessler_1995:
+  - condensation/evaporation as relaxation to equilibrium
+  - autoconversion
+  - accretion
+  - rain evaporation
+  - rain terminal velocity
+"""
+
+module Microphysics
+
+using CLIMA.MoistThermodynamics
+using CLIMA.PlanetParameters
+using ..MicrophysicsParameters
+
+# rain fall speed
+export terminal_velocity_single_drop_coeff
+export terminal_velocity
+
+# rates of conversion between microphysics categories
+export conv_q_vap_to_q_liq
+export conv_q_liq_to_q_rai_acnv
+export conv_q_liq_to_q_rai_accr
+export conv_q_rai_to_q_vap
+
+"""
+    terminal_velocity_single_drop_coeff(ρ)
+
+where:
+  - `ρ` is the density of air
+
+Returns the proportionality coefficient between terminal velocity of an
+individual water drop and the square root of its radius * g.
+"""
+function terminal_velocity_single_drop_coeff(ρ::DT) where {DT<:Real}
+
+    # terminal_vel_of_individual_drop = v_drop_coeff * (g * drop_radius)^(1/2)
+    v_c::DT = sqrt(DT(8/3) / C_drag * (dens_liquid / ρ - DT(1)))
+
+    return v_c
+end
+
+"""
+    terminal_velocity(q_rai, ρ)
+
+where:
+  - `q_rai` - rain water specific humidity
+  - `ρ`     - density of air
+
+Returns the mass weighted average rain terminal velocity assuming
+Marshall Palmer 1948 distribution of rain drops.
+"""
+function terminal_velocity(q_rai::DT, ρ::DT) where {DT<:Real}
+
+    v_c = terminal_velocity_single_drop_coeff(ρ)
+
+    # gamma(9/2)
+    gamma_9_2::DT = DT(11.63)
+
+    lambda::DT = (DT(8) * π * dens_liquid * MP_n_0 / ρ / q_rai)^DT(1/4)
+
+    vel = gamma_9_2 * v_c / DT(6) * sqrt(grav / lambda)
+
+    return vel
+end
+
+
+"""
+    conv_q_vap_to_q_liq(q_sat, q)
+
+where:
+- `q_sat` - PhasePartition at equilibrium
+- `q`     - current PhasePartition
+
+Returns the q_liq tendency due to condensation/evaporation.
+The tendency is obtained assuming a relaxation to equilibrium with
+constant timescale.
+"""
+function conv_q_vap_to_q_liq(q_sat::PhasePartition, q::PhasePartition)
+
+  DT = eltype(q.tot)
+
+  src_q_liq::DT = (q_sat.liq - q.liq) / τ_cond_evap
+
+  if q_sat.ice != DT(0)
+    @show("1-moment bulk microphysics is not defined for snow/ice")
+    #This should be the q_ice tendency due to sublimation/resublimation.
+    #src_q_ice = (q_sat.ice - q.ice) / τ_subl_resubl
+  end
+
+  return src_q_liq
+
+end
+
+
+"""
+    conv_q_liq_to_q_rai_acnv(q_liq)
+
+where:
+- `q_liq` - is the liquid water specific humidity
+
+Returns the q_rai tendency due to collisions between cloud droplets
+(autoconversion) parametrized following Kessler 1995.
+"""
+function conv_q_liq_to_q_rai_acnv(q_liq::DT) where {DT<:Real}
+
+  src_q_rai::DT = max(DT(0), q_liq - q_liq_threshold) / τ_acnv
```

Comment:
```suggestion
  return max(DT(0), q_liq - q_liq_threshold) / τ_acnv
```
and can get rid of extra empty lines

Reply:
done